src/jsn.erl

```diff
@@ -501,13 +512,64 @@ is_subset(_A, _B) ->
 %% conversion API
 %%==============================================================================
 
+-spec as_map(json_term()) -> json_term().
+%%------------------------------------------------------------------------------
+%% @doc convert any JSON objects in the input JSON term into map format
+%%------------------------------------------------------------------------------
+as_map(M) when is_map(M) ->
+    maps:map(fun(Key, Value) when ?IS_JSON_KEY(Key) ->
+                 as_map(Value);
+                (Key, Value) -> erlang:error(badarg, [Key, Value]) 
+             end,
+             M);
+as_map({struct, List}) when is_list(List) ->
+    as_map(List);
+as_map({List}) when is_list(List) ->
+    as_map(List);
+as_map([{struct, H}|T]) when is_list(H) ->
+    [as_map(H) | as_map(T)];
+as_map([{H}|T]) when is_list(H) ->
+    [as_map(H) | as_map(T)];
+as_map([{_K, _V}|_T] = Plist) ->
+    lists:foldl(fun({Key, Value}, Acc) when ?IS_JSON_KEY(Key) ->
```

Comment:
@mrallen1 I was wondering if it would be more efficient, especially for larger proplists, to use

```erlang
M = maps:from_list(Plist),
maps:map(fun(Key, Value) when ?IS_JSON_KEY(Key) ->
             as_map(Value);
            (Key, Value) -> erlang:error(badarg, [Key, Value])
         end,
         M);
```

I haven't compared the performance of the 2 yet; just wondering if you've played with these two approaches before I bother with benchmarking.